sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParameterAssignedTo.cs

```diff
@@ -72,13 +65,68 @@ protected override bool IsAssignmentToParameter(ISymbol symbol)
         {
             var parameterSymbol = symbol as IParameterSymbol;
             var result = parameterSymbol?.RefKind == RefKind.None;
-
             return result;
         }
 
+        protected override bool IsReadBefore(SemanticModel semanticModel, ISymbol symbol, AssignmentExpressionSyntax assignment)
+        {
+            return GetPreviousStatements(assignment).Union(new[] { assignment.Right })
+                .Select(s =>
+                {
+                    var collector = new ParameterAccessCollector(semanticModel, symbol);
+                    collector.Visit(s);
+                    return collector.IsRead;
+                })
+                .Aggregate((isRead, accumulator) => accumulator || isRead);
+        }
+
         protected override SyntaxNode GetAssignedNode(AssignmentExpressionSyntax assignment) => assignment.Left;
 
         protected sealed override GeneratedCodeRecognizer GeneratedCodeRecognizer => Helpers.CSharp.GeneratedCodeRecognizer.Instance;
+
+        /// <summary>
+        /// Returns all statements before the specified statement within the containing method.
+        /// This method recursively traverses all parent blocks of the provided statement.
+        /// </summary>
+        private static IEnumerable<SyntaxNode> GetPreviousStatements(SyntaxNode statement)
+        {
+            var previousStatements = statement.Parent.ChildNodes()
+                .OfType<StatementSyntax>()
+                .TakeWhile(x => x != statement)
+                .Reverse();
+
+            return statement.Parent is StatementSyntax parentStatement
+                ? previousStatements.Union(GetPreviousStatements(parentStatement))
+                : previousStatements;
+        }
+
+        private class ParameterAccessCollector : CSharpSyntaxWalker

```

Comment:
Wouldn't this be simpler without a CSharpSyntaxWalker? Just something like `node.Descendants<Identifier>.Any(IsRead)`?

Reply:
Actually this would have better to be on line 80...

Reply:
That's ok, I've changed it to use just LINQ.